XenAdmin/Alerts/Types/ClientUpdateAlert.cs

```diff
@@ -134,17 +135,64 @@ public static void DownloadAndInstallNewClient(ClientUpdateAlert updateAlert, IW
             {
                 Process.Start(outputPathAndFileName);
                 log.DebugFormat("Update {0} found and install started", updateAlert.Name);
-                downloadAndInstallClientAction.ReleaseInstaller();
+                downloadAndInstallClientAction.ReleaseDownloadedContent();
                 Application.Exit();
             }
             catch (Exception e)
             {
                 log.Error("Exception occurred when starting the installation process.", e);
-                downloadAndInstallClientAction.ReleaseInstaller(true);
+                downloadAndInstallClientAction.ReleaseDownloadedContent(true);
 
                 using (var dlg = new ErrorDialog(Messages.UPDATE_CLIENT_FAILED_INSTALLER_LAUNCH))
                     dlg.ShowDialog(parent);
             }
         }
+
+        public static void DownloadSource(IWin32Window parent)
+        {
+            string outputPathAndFileName;
+            using (var saveSourceDialog = new SaveFileDialog())
+            {
+                saveSourceDialog.FileName = BrandManager.BrandConsoleNoSpace + "-source.zip";
+                saveSourceDialog.DefaultExt = "zip";
+                saveSourceDialog.Filter = "(*.zip)|*.zip|All files (*.*)|*.*";
+                saveSourceDialog.InitialDirectory = Win32.GetKnownFolderPath(Win32.KnownFolders.Downloads);
+
+                if (saveSourceDialog.ShowDialog() != DialogResult.OK)
+                {
+                    return;
+                }
+                outputPathAndFileName = saveSourceDialog.FileName;
+            }
+
+            var clientVersion = Updates.ClientVersions.FirstOrDefault();

```

Comment:
This method is also called from the item under the help menu when alerts are absent, in which case it won't download the source. Maybe make the clientVersion a parameter and set it accordingly in the item click handlers.

Reply:
It will as the clientVersions is the pre-processed list of updates, it only becomes an update alert if the installed version < update version